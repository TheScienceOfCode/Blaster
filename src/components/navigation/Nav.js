import React from "react";
import { Route, Switch } from "react-router-dom";

import Home from "../../pages/home/Home";
import About from "../../pages/about/About";
import Contact from "../../pages/contact/Contact";
import Articles from "../../pages/articles/Articles";

const Nav = ({location}) => (
  <Switch key={location.key}>
    <Route exact path="/"   location={location} component={Home} />
    <Route path="/articles" location={location} component={Articles} />
    <Route path="/about"     location={location} component={About} />
    <Route path="/contact"  location={location} component={Contact} />
  </Switch>
);

export default Nav;
