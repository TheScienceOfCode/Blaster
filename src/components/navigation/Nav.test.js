import React from 'react'
import renderer from 'react-test-renderer'
import Nav from './Nav.js';
import MockRouter from 'react-mock-router';

const location = { key: ''};

describe('nav snapshot test', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<MockRouter location={location}><Nav location={location}/></MockRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
