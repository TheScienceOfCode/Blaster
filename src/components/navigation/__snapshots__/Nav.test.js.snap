// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nav snapshot test renders correctly 1`] = `null`;
